elfpy/agents/policies/random_agent.py

```diff
@@ -65,24 +65,27 @@ def open_short_with_random_amount(self, market: HyperdriveMarket, wallet: Wallet
         maximum_trade_amount = market.get_max_short_for_account(wallet.balance.amount)
         if maximum_trade_amount <= WEI:
             return []
+
         initial_trade_amount = FixedPoint(
             self.rng.normal(loc=float(self.budget) * 0.1, scale=float(self.budget) * 0.01)
         )
         # WEI <= trade_amount <= max_short
         trade_amount = max(WEI, min(initial_trade_amount, maximum_trade_amount))
         # return a trade using a specification that is parsable by the rest of the sim framework
+
         return [
             Trade(
                 market=MarketType.HYPERDRIVE,
                 trade=HyperdriveMarketAction(
                     action_type=MarketActionType.OPEN_SHORT,
                     trade_amount=trade_amount,
+                    slippage_tolerance=self.slippage_tolerance,
                     wallet=wallet,
                 ),
             )
         ]
 
-    def close_random_short(self, wallet: Wallet) -> list[Trade]:
+    def close_random_short(self, wallet: Wallet) -> list[Trade[HyperdriveMarketAction]]:
```

Comment:
this checks if there's a HyperdriveMarketAction inside the Trade?

Reply:
it assigns a type to the action for the trade. Before we would generically use `Any`. Now we can say this is a `HyperdriveMarketAction` trade, versus a `BorrowMarketAction` trade, for example.

Reply:
yeah!  the `trade` attribute of Trade was an `Any` before.  Now it's generic so we can specify what's in there.